pkg/pkcs/pkcs.go

```diff
@@ -25,13 +25,9 @@ func generatePublicKey(w io.Writer, key *rsa.PrivateKey) error {
 }
 
 func generatePrivateKey(w io.Writer, key *rsa.PrivateKey) error {
-	pkcs8PrivateKey, err := x509.MarshalPKCS8PrivateKey(key)
-	if err != nil {
-		return fmt.Errorf("cannot marshal the private key to PKCS8: %w", err)
-	}
 	if err := pem.Encode(w, &pem.Block{
 		Type:  "RSA PRIVATE KEY",
-		Bytes: pkcs8PrivateKey,
+		Bytes: x509.MarshalPKCS1PrivateKey(key),
```

Comment:
this also needs to changed because of a wrong format, Kubernetes API will throw errors.